src/engine/helper/pka_helper.c

```diff
@@ -181,6 +181,103 @@ static void free_operand(pka_operand_t *operand)
     free(operand);
 }
 
+/* ECC related helpers */
+
+static void make_ecc_operand(PKA_ULONG     *bn_buf_ptr,
+                             uint32_t       buf_len,
+                             uint32_t       buf_max_len,
+                             uint8_t        big_endian,
+                             pka_operand_t *operand)
+{
+    if (operand == NULL)
+        return;
+
+    if (!bn_buf_ptr || (buf_max_len == 0))
+        return;
+
+    memset(operand, 0, sizeof(pka_operand_t));
+    operand->big_endian = big_endian;
+
+    // Now init the operand buffer
+    operand->buf_ptr    = malloc(buf_max_len);
+    operand->buf_len    = buf_max_len;
+    memset(operand->buf_ptr, 0, buf_max_len);
+
+    // Now fill the operand buf.
+    operand_byte_copy(operand, (uint8_t *) bn_buf_ptr, buf_len);
+
+}
+static void bignum_to_ecc_operand(pka_bignum_t  *bignum,
+                                  pka_operand_t *operand)
+{
+    if (operand == NULL)
+        return;
+
+    uint32_t byte_len, byte_max_len;
+
+    if (bignum)
+    {
+        byte_len     = bignum->top  * PKA_BYTES;
+        byte_max_len = bignum->dmax * PKA_BYTES;
+
+        make_ecc_operand(bignum->d, byte_len, byte_max_len, 0, operand);
+    }
+    return;
+}
+
+static void malloc_ecc_operand(uint32_t       buf_len,
+                               pka_operand_t *operand)
+{
+    if (operand == NULL)
+        return;
+
+    memset(operand, 0, sizeof(pka_operand_t));
+    operand->buf_ptr    = malloc(buf_len);
+    memset(operand->buf_ptr, 0, buf_len);
+    operand->buf_len    = buf_len;
+    operand->actual_len = 0;
+
+    return;
+}
+
+static ecc_point_t *malloc_ecc_point(uint32_t x_buf_len,
+                                     uint32_t y_buf_len)
+{
+    ecc_point_t *point;
+
+    point = malloc(sizeof(ecc_point_t));
+    memset(point, 0, sizeof(ecc_point_t));
+
+    malloc_ecc_operand(x_buf_len, &point->x);
+    malloc_ecc_operand(y_buf_len, &point->y);
+
+    return point;
+}
+
+static void free_ecc_operand(pka_operand_t *operand)
+{
+    uint8_t *buf_ptr;
+
+    if (operand == NULL)
+        return;
+
+    buf_ptr = operand->buf_ptr;
+
+    if (buf_ptr != NULL)
+        free(buf_ptr);
```

Comment:
free(operand)?

Reply:
`ecc_point_t `has x,y variables declared as `pka_operand_t`, hence these operands are not dynamically created.